### PR TITLE
Accept Ligature Caret Value Format 3

### PR DESCRIPTION
**Pull request by felipesanches**
(issue #178)

captura de tela 2019-01-18 as 12 57 29

See also: https://github.com/googlefonts/fontbakery/issues/2268

### CODE REVIEW

**Comment by davelab6** [comment]
Thank you @felipesanches :)

**Comment by behdad** [comment]
Funny what that piece of code does... or doesn't:
```
    // Parse caret values table 
    for (unsigned j = 0; j < caret_count; ++j) { 
      subtable.set_offset(lig_glyphs[i] + caret_value_offsets[j]); 
      uint16_t caret_format = 0; 
      if (!subtable.ReadU16(&caret_format)) { 
        return Error("Can't read caret values table %d in glyph %d", j, i); 
      } 
      // TODO(bashi): We only support caret value format 1 and 2 for now 
      // because there are no fonts which contain caret value format 3 
      // as far as we investigated. 
      if (caret_format == 0 || caret_format > kMaxCaretValueFormat) { 
        return Error("bad caret value format: %u", caret_format); 
      } 
      // CaretValueFormats contain a 2-byte field which could be 
      // arbitrary value. 
      if (!subtable.Skip(2)) { 
        return Error("Bad caret value table structure %d in glyph %d", j, i); 
      } 
    } 
```
It doesn't do anything other than checking format number.  Why do people insist on giving the false sense to font developers that ots is checking anything meaningful in their GSUB/GPOS/GDEF tables?!!

**Comment by davelab6** [comment]
LOL

I guess the chrome team engineers who started OTS had their heart in the
right place :)

**Comment by khaledhosny** [comment]
At least it makes sure the table do actually contains two bytes for the coordinate of each caret. Just allowing format 3 is not enough as it has more data. But since this code does not do much sensitization, one just need to check that the data is there.

File: src/gdef.cc [hunk 1]
```diff
@@ -167,9 +165,6 @@ bool OpenTypeGDEF::ParseLigCaretListTable(const uint8_t *data, size_t length) {
       if (!subtable.ReadU16(&caret_format)) {
         return Error("Can't read caret values table %d in glyph %d", j, i);
       }
-      // TODO(bashi): We only support caret value format 1 and 2 for now
-      // because there are no fonts which contain caret value format 3
-      // as far as we investigated.
       if (caret_format == 0 || caret_format > kMaxCaretValueFormat) {
         return Error("bad caret value format: %u", caret_format);
       }
```
Comment (khaledhosny) [inline src/gdef.cc]: You need to modify the `subtable.Skip(2)` below for format 3 as it has more data: https://docs.microsoft.com/en-us/typography/opentype/spec/gdef#caretvalue-format-3

**Comment by khaledhosny** [comment]
Funnily, it was ligature carets that eventually lead me to fork OTS since it was rejecting my perfectly valid fonts because they had ligature caret. It was my very first bug fix: #1.

**Comment by davelab6** [comment]
Haha! Is there a Chromium bug to render using them?

**Pull request merged**